base/common/src/main/java/org/dogtagpki/nss/NSSExtensionGenerator.java

```diff
@@ -197,20 +200,35 @@ public SubjectKeyIdentifierExtension createSKIDExtension(X509Key subjectKey) thr
 
         logger.info("Creating SKID extension:");
 
-        byte[] bytes;
+        boolean critical = false;
+        byte[] bytes = null;
+
+        List<String> values = Arrays.asList(subjectKeyIdentifier.split(","));
+        for (String value : values) {
+            value = value.trim();
+            logger.info("- " + value);
+
+            if (value.equals("critical")) {
+                critical = true;
+                continue;
+            }
 
-        if (subjectKeyIdentifier.equals("hash")) {
-            logger.info("- hash");
-            bytes = CryptoUtil.generateKeyIdentifier(subjectKey.getKey());
+            if (value.equals("hash")) {
+                bytes = CryptoUtil.generateKeyIdentifier(subjectKey.getKey());
+                continue;
+            }
 
-        } else {
-            throw new Exception("Unsupported subjectKeyIdentifier: " + subjectKeyIdentifier);
+            try {
+                bytes = HEX_FORMAT.parseHex(value);
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Unsupported subject key identifier: " + value + ": " + e.getMessage(), e);
+            }
```

Comment:
If I have understood correctly the OpenSSL specification `critical` has to be be the first element and cannot be in other positions. Additionally, `subjectKeyIdentifier` is a _string_ and not a _multi-valued_. If this is correct then you cannot go with the loop.

Of course we can diverge from OpenSSL but we should write how to use `sky_id` properly